shared/js/views.jsx

```diff
@@ -599,22 +599,54 @@ loop.shared.views = (function(_, mozL10n) {
       isLoading: React.PropTypes.bool.isRequired,
       mediaType: React.PropTypes.string.isRequired,
       posterUrl: React.PropTypes.string,
+      shouldRenderRemoteCursor: React.PropTypes.bool,
       // Expecting "local" or "remote".
       srcMediaElement: React.PropTypes.object
     },
 
+    getInitialState: function() {
+      return {
+        videoLetterboxing: {
+          left: 0,
+          top: 0
+        },
+        streamVideoHeight: 0,
+        streamVideoWidth: 0
+      };
+    },
+
     componentDidMount: function() {
       if (!this.props.displayAvatar) {
         this.attachVideo(this.props.srcMediaElement);
       }
     },
 
+    componentWillUnmount: function() {
+      var videoElement = this.getDOMNode().querySelector("video");
+      if (!this.props.shouldRenderRemoteCursor || !videoElement) {
+        return;
+      }
+
+      videoElement.removeEventListener("timeupdate", this.handleVideoUpdate);
```

Comment:
Did you mean to remove `"loadeddata"`? Although with `timeupdate`, you might be able to detect stream and element size changes?